### PR TITLE
mgr/dashboard: Refactor RGW backend

### PR DESCRIPTION
**Pull request by votdev**
- Do some polishing in the docs.
- Do some code cleanup in RgwClient::_load_settings().
- Refactor RgwClient::is_service_online() method. The system is considered as online if the response structure is valid. The response content itself is not validated in this case.

Signed-off-by: Volker Theile <vtheile@suse.com>

### CODE REVIEW

File: src/pybind/mgr/dashboard/controllers/rgw.py [hunk 1]
```diff
@@ -80,7 +81,7 @@ def __call__(self, path, **params):
                 data = cherrypy.request.body.read()
 
             return rgw_client.proxy(method, path, params, data)
-        except RequestException as e:
+        except (NoCredentialsException, RequestException) as e:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/controllers/rgw.py]: @p-na `NoCredentialsException` is only used in the rgw controller. Would it make sense to make `NoCredentialsException` inherent from `RequestException`?

Comment (votdev) [reply src/pybind/mgr/dashboard/controllers/rgw.py]: Makes sense

File: doc/mgr/dashboard.rst [hunk 2]
```diff
@@ -156,7 +156,7 @@ command.
 The credentials of an existing user can also be obtained by using
 `radosgw-admin`::
 
-  $ radosgw-admin user info --uid=<user id>
+  $ radosgw-admin user info --uid=<user_id>
 
 Finally, provide the credentials to the dashboard module::
```
Comment (sebastian-philipp) [inline doc/mgr/dashboard.rst]: Please add `$ ceph dashboard set-rgw-api-secret-key <user id>` to the section below ⬇⬇⬇

**Comment by votdev** [comment]
@sebastian-philipp Your comment has been adressed

**Pull request merged**